servers/visual_server.cpp

```diff
@@ -333,10 +333,21 @@ RID VisualServer::get_white_texture() {
 // Resulting 2D vector in range [-1, 1]
 // See http://jcgt.org/published/0003/02/01/ for details
 Vector2 VisualServer::norm_to_oct(const Vector3 v) {
-	const float invL1Norm = (1.0f) / (Math::absf(v.x) + Math::absf(v.y) + Math::absf(v.z));
+	const float L1Norm = Math::absf(v.x) + Math::absf(v.y) + Math::absf(v.z);
+
+	// NOTE: we support NAN values here, but we use them because our input is infinitely small.
+	// This means we can safely set the value to zero and avoid the division by zero, this prevents regressions
+	// So basically the compressor can take valid values, or a nan (a nan is used as a special flag for zero explicitly)
+	// Remember 1 / X -> graphs to close to zero which in our case is fine to be zero.
+	// Reference: https://www.wolframalpha.com/input/?i2d=true&i=Divide%5B1%2Cx%5D
+	if (Math::is_zero_approx(L1Norm)) {
+		WARN_PRINT_ONCE("Octahedral compression cannot be used to compress a zero-length vector, please use normalized normal values or disable octahedral compression")
+		return Vector2(NAN, NAN);
```

Comment:
Let's just return ``Vector2()`` so it is clear that we don't care about the return value (even though we know this will decompress to ``Vector3(0,0,1)``